command/agent/auth/auth.go

```diff
@@ -167,6 +170,11 @@ func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) error {
 			clientToUse, err = am.(AuthMethodWithClient).AuthClient(ah.client)
 			if err != nil {
 				ah.logger.Error("error creating client for authentication call", "error", err, "backoff", backoff)
+
+				if ah.exitOnError {
+					return err
+				}
+
 				backoffOrQuit(ctx, backoff)
 				metrics.IncrCounter([]string{"agent", "auth", "failure"}, 1)
```

Comment:
It might be good to increment this counter before returning, so that the failures are counted.